networksurvey/src/main/java/com/craxiom/networksurvey/fragments/GrpcConnectionFragment.java

```diff
@@ -214,25 +215,63 @@ private synchronized void updateUiState(ConnectionState connectionState)
     }
 
     /**
-     * Read the connection values from the UI, and then start the {@link GrpcConnectionService} and pass it the
-     * parameters so it can establish the connection to the gRPC server.
+     * Validates the user-specified server connection parameters
+     * @param host Specified host name. All non empty strings are 'valid'.
+     * @param port Specified port number. All numeric numbers between 0 and 65535 are 'valid'.
+     * @param deviceName Specified device name. Not validated.
+     * @return True if all parameters are valid according to the above criteria.
      */
-    private void connectToGrpcServer()
+    private boolean areConnectionParamsValid(String host, String port, String deviceName)
```

Comment:
This method both validates the values and also updates the instance variables.  I think it might be better to have this method just validate the values, and handle updating the instance variables somewhere else.